### ISSUE TITLE
Support multiple projects in one VSCode workspace

### ISSUE THREAD

**Issue by cocreature** [digital-asset/daml#3867]
Currently our VSCode extension only works properly if you open VSCode in the folder where your `daml.yaml` is. Apart from terrible error messages if you screw this up (see #3300 for a suggestion for making it slightly better), this will simply become infeasible as people move towards more fine-grained packages which becomes crucial for upgrading.

Therefore, we should support this natively. My understanding so far was that this needs to be supported by the LSP server but it turns out that is actually not necessary. VSCode makes it really easy to spin up multiple servers and automatically select one based on a file prefix. See https://github.com/microsoft/vscode-extension-samples/blob/926c5c045e3274a0572f4a07ca0adf6d2fdbd949/lsp-multi-server-sample/client/src/extension.ts for an example. There are two options here: We could either require users to use multifolder workspaces like in this example or we can simply detect the `daml.yaml` belonging to a file and use that to determine which client we need to use. The latter seems simple enough and should be more convenient for users so I would suggest to go with that.

**Comment by cocreature** [comment]
Two give some context for the following, we assume that we have the following directory structure
```
- root
  - proja
     - daml.yaml
  - projb
     - daml.yaml
```

After having spend 2 days or so on this, this has turned out to be significantly more complex than I expected it to be:

1. VSCode doesn’t like the fact that we register commands multiple times. The recommended way of working around this seems to be to include UUIDs in commands but that is annoying and I expect that there are probably others poblems with trying to host multiple LSP servers in one workspace. In addition to that it adds a fair amount of complexity to the extension which is significantly harder to test than the server (in fact it’s not tested at all atm). Therefore I don’t think my original idea of trying to solve this by starting multiple servers from the extension is desirable.
2. If we don’t solve this in the extension, we have to solve it in the server. There are several issues here:
    1. If we solve it in the server it is unclear which SDK version should be used since the server will be started in `root` where there is no `daml.yaml` not in `proja` or `projb`. A hacky workaround for that is to have a dummy `daml.yaml` in `root`. Of course this will go wrong if you actually use different SDK versions but I think that’s somewhat acceptable. If you’ve upgraded, you don’t need to see the sources of older packages anymore.
    2. The good news is that the `GhcSession` rule is setup for handling multiple sessions. We can reasonably easy set it up such that this work.
    3. The bad news is that a ton of stuff relies on the current working directory. We can fix this for `GhcSession` reasonably easily since it takes a file argument so we can use that t locate the project root.
    4. We have similar issues for `initPackageDb` which can be solved but it’s a bit annoying to do.
    5. The biggest problem appears to be `GeneratePackageMap`. This rule does not depend on the file which means we have no way to find the project root and are therefore completely screwed. I don’t see any way of fixing this without changing this. This should be possible but it’s probably a fair amount of effort.

**Comment by cocreature** [comment]
Fixed in #4445
